core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/ClusterHostServiceRunner.java

```diff
@@ -458,8 +466,7 @@ private void saveGatewayPillar(GatewayConfig gatewayConfig, Cluster cluster, Map
         gateway.put("kerberos", kerberosConfig != null);
 
         boolean ambariBlueprint = blueprintService.isAmbariBlueprint(cluster.getBlueprint());
-        List<String> serviceNames = ambariBlueprint ? ExposedService.getAllServiceNameForAmbari() : ExposedService.getAllServiceNameForCM();
-        Map<String, List<String>> serviceLocation = componentLocator.getComponentLocation(cluster, serviceNames);
+        Map<String, List<String>> serviceLocation = getServiceLocations(cluster, ambariBlueprint);
```

Comment:
Maybe worth renaming the variable to `serviceLocations` to be consistent with the method name (and the change above).

Reply:
fixed

Reply:
Perfect.